### ISSUE TITLE
Unable to build on Windows

### ISSUE THREAD

**Issue by xobs** [nicolodavis/boardgame.io#647]
I'm running Node v14.0.0 and trying to use `boardgame.io`.

The `README.md` file says to install I should run `npm install --save boardgame.io`, however that returns an error:

```
[9:58:27 AM] D:/Code/mj/boardgame.io> npm install --save boardgame.io
npm ERR! code ENOSELF
npm ERR! Refusing to install package with name "boardgame.io" under a package
npm ERR! also called "boardgame.io". Did you name your project the same
npm ERR! as the dependency you're installing?
npm ERR!
npm ERR! For more information, see:
npm ERR!     <https://docs.npmjs.com/cli/install#limitations-of-npms-install-algorithm>

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\smcro\AppData\Roaming\npm-cache\_logs\2020-04-25T01_59_36_055Z-debug.log
[9:59:36 AM] D:/Code/mj/boardgame.io>
```

The `README.md` also says I should run `npm install; npm run build`, however this also fails:

```
[9:54:45 AM] D:/Code/mj/boardgame.io> npm install

> deasync@0.1.15 install D:\Code\mj\boardgame.io\node_modules\deasync
> node ./build.js


D:\Code\mj\boardgame.io\node_modules\deasync>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
  deasync.cc
  win_delay_load_hook.cc
     Creating library D:\Code\mj\boardgame.io\node_modules\deasync\build\Release\deasync.lib and object D:\Code\mj\boardgame.io\node_modules\deasync\build\Release\deasync.exp
  deasync.vcxproj -> D:\Code\mj\boardgame.io\node_modules\deasync\build\Release\\deasync.node
Installed in `D:\Code\mj\boardgame.io\node_modules\deasync\bin\win32-x64-node-14\deasync.node`

> husky@1.3.1 install D:\Code\mj\boardgame.io\node_modules\husky
> node husky install

husky > setting up git hooks
husky > done

> core-js@2.6.9 postinstall D:\Code\mj\boardgame.io\node_modules\core-js
> node scripts/postinstall || echo "ignore"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon:
> https://opencollective.com/core-js
> https://www.patreon.com/zloirock

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)


> docsify@4.11.2 postinstall D:\Code\mj\boardgame.io\node_modules\docsify
> opencollective-postinstall

Thank you for using docsify!
If you rely on this package, please consider supporting our open collective:
> https://opencollective.com/docsify/donate


> nodemon@1.19.2 postinstall D:\Code\mj\boardgame.io\node_modules\nodemon
> node bin/postinstall || exit 0

Love nodemon? You can now support the project via the open collective:
 > https://opencollective.com/nodemon/donate

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

added 1633 packages from 847 contributors and audited 890206 packages in 95.93s

4 packages are looking for funding
  run `npm fund` for details

found 847 vulnerabilities (845 low, 2 moderate)
  run `npm audit fix` to fix them, or `npm audit` for details
[9:56:41 AM] D:/Code/mj/boardgame.io> npm run build

> boardgame.io@0.39.9 build D:\Code\mj\boardgame.io
> cross-env BABEL_ENV=rollup rollup --config rollup.config.js


packages/client.ts, packages/core.ts, packages/debug.js, packages/react.ts, packages/react-native.ts, packages/ai.ts, packages/plugins.ts, packages/master.ts, packages/multiplayer.ts, packages/internal.ts → dist/esm, dist/cjs...
[!] (plugin babel) Error: Cannot find module '@babel/compat-data/corejs3-shipped-proposals'
Require stack:
- D:\Code\mj\boardgame.io\node_modules\@babel\preset-env\lib\polyfills\corejs3\usage-plugin.js
- D:\Code\mj\boardgame.io\node_modules\@babel\preset-env\lib\index.js
- D:\Code\mj\boardgame.io\node_modules\@babel\core\lib\config\files\plugins.js
- D:\Code\mj\boardgame.io\node_modules\@babel\core\lib\config\files\index.js
- D:\Code\mj\boardgame.io\node_modules\@babel\core\lib\index.js
- D:\Code\mj\boardgame.io\node_modules\rollup-plugin-babel\dist\rollup-plugin-babel.cjs.js
- D:\Code\mj\boardgame.io\rollup.config.js
- D:\Code\mj\boardgame.io\node_modules\rollup\dist\bin\rollup
packages\debug.js
Error: Cannot find module '@babel/compat-data/corejs3-shipped-proposals'
Require stack:
- D:\Code\mj\boardgame.io\node_modules\@babel\preset-env\lib\polyfills\corejs3\usage-plugin.js
- D:\Code\mj\boardgame.io\node_modules\@babel\preset-env\lib\index.js
- D:\Code\mj\boardgame.io\node_modules\@babel\core\lib\config\files\plugins.js
- D:\Code\mj\boardgame.io\node_modules\@babel\core\lib\config\files\index.js
- D:\Code\mj\boardgame.io\node_modules\@babel\core\lib\index.js
- D:\Code\mj\boardgame.io\node_modules\rollup-plugin-babel\dist\rollup-plugin-babel.cjs.js
- D:\Code\mj\boardgame.io\rollup.config.js
- D:\Code\mj\boardgame.io\node_modules\rollup\dist\bin\rollup
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1014:15)
    at Function.Module._load (internal/modules/cjs/loader.js:884:27)
    at Module.require (internal/modules/cjs/loader.js:1074:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (D:\Code\mj\boardgame.io\node_modules\@babel\preset-env\lib\polyfills\corejs3\usage-plugin.js:10:55)
    at Module._compile (internal/modules/cjs/loader.js:1185:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1205:10)
    at Module.load (internal/modules/cjs/loader.js:1034:32)
    at Function.Module._load (internal/modules/cjs/loader.js:923:14)
    at Module.require (internal/modules/cjs/loader.js:1074:19)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! boardgame.io@0.39.9 build: `cross-env BABEL_ENV=rollup rollup --config rollup.config.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the boardgame.io@0.39.9 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\smcro\AppData\Roaming\npm-cache\_logs\2020-04-25T01_56_59_999Z-debug.log
[9:57:00 AM] D:/Code/mj/boardgame.io> node --version
v14.0.0
[9:58:27 AM] D:/Code/mj/boardgame.io>
```

This is a fresh install of node, and I have removed `npm` from my home directory in order to ensure there are no global modules installed.

**Comment by ksjogo** [comment]
```
 D:/Code/mj/boardgame.io> npm install --save boardgame.io
npm ERR! code ENOSELF
npm ERR! Refusing to install package with name "boardgame.io" under a package
npm ERR! also called "boardgame.io". Did you name your project the same
npm ERR! as the dependency you're installing?
```
You will  have to rename your project folder to add boardgame.io as dependency

**Comment by xobs** [comment]
I wanted to try the examples first.  Do you not need to install it?

Also, trying the examples failed for some reason (see the second command).

**Comment by ksjogo** [comment]
Did you git clone the boardgame.io repo?
Then you don't need to/can't do `npm install --save boardgame.io` because you already got all the files.

**Comment by xobs** [comment]
For the first problem,

Yes, I did clone the git repo.  I wasn't sure if `npm install --save` did something like save `boardgame.io` to my user profile directory, ensuring it was "installed".  I'll skip that for now.

For the second problem, I'm able to get a successful build after running `npm install '@babel/compat-data'`. So that does seem like a required installation step.

**Comment by delucis** [comment]
Hi @xobs, we’ve just merged a PR to fix Windows builds (#657).

To run the examples, you should clone the repository and run the following in the directory.

```sh
# install the dependencies boardgame.io needs
npm install
# start the examples server
npm start
```

The `npm install --save boardgame.io` is for adding the package to your own project, it can’t be run inside the boardgame.io directory.

I’m going to close this, but let us know if you run into any further issues!
